saleor/graphql/product/tests/mutations/test_product_variant_update.py

```diff
@@ -2533,3 +2533,117 @@ def test_update_product_variant_can_not_turn_off_preorder(
     assert data["sku"] == sku
     assert data["preorder"]["globalThreshold"] == variant.preorder_global_threshold
     assert data["preorder"]["endDate"] is None
+
+
+@patch("saleor.plugins.manager.PluginsManager.product_variant_updated")
+@patch("saleor.plugins.manager.PluginsManager.product_variant_metadata_updated")
+def test_update_product_variant_with_metadata(
```

Comment:
Can you add additional tests to cover situations like:
- metadata/private metadata input is empty when:
  - variant.metadata/private_metada is also empty - without metadata webhook event
  - variant.metadata/private_metadata contains different keys - without metadata webhook event
- metadata/private metadata input has key:"a", value:"b" when:
  - variant.metadata/private metadata has the same key&value - without metadata webhook event
  - variant.metdata/private metadata has the same key but different value - with metadata webhook event